lib/adapter.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+const Pensuer = require('penseur');
+const Hoek = require('hoek');
+const Path = require('path');
+
+const Adapter = function (params) {
+
+    this.params = params || {};
+    if (!this.params.database) {
+        throw new Error('Database Connection params should be set');
+    }
+};
+
+Adapter.prototype.getTemplatePath = function () {
+
+    return Path.join(__dirname, 'migrationTemplate.js');
+};
+
+Adapter.prototype.connect = function (callback) {
+
+    const self = this;
+    const settings = Hoek.clone(self.params.database.connection);
+    this.db = new Pensuer.Db(self.params.database.name, settings);
+    //ensure migrations table exists in the database
+    if (self.params.tables.indexOf('migration') === -1) {
```

Comment:
I was passing in the 'tables' as I was under the impression that Penseur needed to know what tables to connect to. (seems to be used everywhere Penseur is). Is this unnecessary overhead?